crates/ethereum/engine/src/service.rs

```diff
@@ -49,19 +54,36 @@ where
     Client: HeadersClient + BodiesClient + Clone + Unpin + 'static,
 {
     /// Constructor for `EthService`.
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
         chain_spec: Arc<ChainSpec>,
         client: Client,
-        to_tree: Sender<FromEngine<BeaconEngineMessage<EthEngineTypes>>>,
-        from_tree: UnboundedReceiver<EngineApiEvent>,
         incoming_requests: UnboundedReceiverStream<BeaconEngineMessage<EthEngineTypes>>,
         pipeline: Pipeline<DB>,
         pipeline_task_spawner: Box<dyn TaskSpawner>,
+        provider: ProviderFactory<DB>,
+        blockchain_db: BlockchainProvider<DB>,
```

Comment:
mental note here, that this type will change,
we could also operate on the provider traits here, but for now these seems easier
The inmemory state could then be obtained from this provider as well